github-api-call-inventory/github-api-call-inventory.ipynb

```diff
@@ -0,0 +1,765 @@
+    "            raise TypeError(f\"unsupported operand type(s) for *: {type(self)!r} and {type(num)!r}\")\n",
+    "        return Task(*[num * getattr(self, field.name) for field in fields(self.__class__)])\n",
+    "    \n",
+    "    __rmul__ = __mul__\n",
+    "    \n",
+    "    def __add__(self, obj):\n",
+    "        if not isinstance(obj, Task) and not isinstance(obj, int):\n",
+    "            raise TypeError(f\"unsupported operand type(s) for +: {type(self)!r} and {type(obj)!r}\")\n",
+    "        if isinstance(obj, int) and obj != 0:\n",
+    "            raise TypeError(f\"adding {obj!r} to an {type(self)!r} is not supported\")\n",
+    "        if isinstance(obj, int) and obj == 0:\n",
+    "            obj = Task(*[0] * len(fields(self.__class__)))\n",
+    "        return Task(*[(getattr(self, field.name) + getattr(obj, field.name)) for field in fields(self.__class__)])\n",
+    "    \n",
+    "    __radd__ = __add__"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "464ee1d0",
+   "metadata": {},
+   "source": [
+    "## Estimating the number of API calls"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "bd8ae47b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from collections import namedtuple\n",
+    "\n",
+    "metrics = [\n",
+    "    \"prs_per_hour_per_org\", \n",
+    "    \"total_is_project_private\", \n",
+    "    \"total_is_user_approved\", \n",
+    "    \"total_load_packit_yaml\", \n",
+    "    \"total_set_status\",\n",
+    "    \"grand_total\",\n",
+    "]\n",
+    "\n",
+    "class APICallMetrics(namedtuple(\"APICallMetrics\", metrics)):\n",
+    "    def __repr__(self):\n",
+    "        return \"\\n\".join(f\"{field} = {getattr(self, field)}\" for field in self._fields)\n",
+    "    \n",
+    "def github_api_calls(\n",
+    "    # The number of chroots configured to be built in Copr\n",
+    "    chroots=3, \n",
+    "    # GitHub server-to-server rate limit per hour\n",
+    "    rate_limit=5000,\n",
+    "    # Calls to load .packit.yaml.\n",
+    "    # The current implementation cycles through a list of 7 files.\n",
+    "    # If the config file is .packit.yaml, this is loaded from 2 calls.\n",
+    "    # If the config file is packit.yaml, it's loaded from 5 calls.\n",
+    "    # The +1 call is to try to find the specfile path.\n",
+    "    load_packit_yaml=2+1,\n",
+    "    # Whether builds are enabled\n",
+    "    builds_enabled=True, \n",
+    "    # Whether tests are enabled\n",
+    "    tests_enabled=True, \n",
+    "    # Load packit.yaml once during the pipeline run\n",
+    "    load_packit_yaml_once=False, \n",
+    "    # Set all flags (including test ones) when the build status changes\n",
+    "    all_flags=True,\n",
+    "    # Check if the repo is private only in the beginning\n",
+    "    is_private_once=False,\n",
+    "    # Check if a user is allowed to trigger the pipeline only once\n",
```

Comment:
maybe I am mistaken, but I think the user check is done only once

Reply:
Right, Copr, Koji and TestingFarm event's [don't trigger this check](https://github.com/packit/packit-service/blob/bc1f66bb82aa7c013bde8b8e670ce7e02f55de41/packit_service/worker/allowlist.py#L237). Thanks for pointing it out!